api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseDaoImpl.java

```diff
@@ -440,6 +442,73 @@ protected boolean containsAlias(Iterator<CriteriaImpl.Subcriteria> subcriteriaIt
 		});
 	}
 	
+	protected void handleLocationReference(Criteria criteria, ReferenceParam locationReference) {
+		if (locationReference != null) {
+			criteria.createAlias("location", "l");
+			
+			if (locationReference.getChain() != null) {
+				switch (locationReference.getChain()) {
+					case Location.SP_ADDRESS_CITY:
+						criteria.add(ilike("l.cityVillage", locationReference.getValue(), MatchMode.START));
```

Comment:
Can we simplify the `ilike()` construction to just `propertyLike("l.cityVillage", locationReference.getValue())`?

Reply:
@ibacher Then how will I add this to criteria? Should I have a list ? 
Something like this

```java
List<Optional<Criterion>> criterionList = new ArrayList<>();
						
						for (String token : StringUtils.split(participantReference.getValue(), " \t,")) {
							criterionList.add(propertyLike("pn.givenName", token));
							criterionList.add(propertyLike("pn.middleName", token));
							criterionList.add(propertyLike("pn.familyName", token));
						}
```